pkg/cloud/instance.go

```diff
@@ -159,6 +167,22 @@ func (c *client) GetOrCreateVMInstance(
 
 	if len(csMachine.Spec.AffinityGroupIds) > 0 {
 		p.SetAffinitygroupids(csMachine.Spec.AffinityGroupIds)
+	} else if !(strings.ToLower(csMachine.Spec.Affinity) == "no" || csMachine.Spec.Affinity == "") {
+		ownerRef := csCtrlrUtils.GetManagementOwnerRef(machine)
+		if ownerRef == nil {
+			return errors.Errorf("Could not find management owner reference for %s/%s",
+				csMachine.Namespace, csMachine.Name)
+		}
+		name := fmt.Sprintf("Affinity-%s", ownerRef.UID)
+		affinityType := AffinityGroupType
+		if strings.ToLower(csMachine.Spec.Affinity) == "anti" {
```

Comment:
Following from Rob's previous feedback, let's avoid literals in code.

Reply:
If the literal is only used once, I'm okay with keeping it as a literal.